spyder/widgets/ipythonconsole/namespacebrowser.py

```diff
@@ -190,6 +190,13 @@ def _handle_data_message(self, msg):
             self.refresh_from_pdb(pdb_state)
 
     # ---- Private API (overrode by us) ----------------------------
+    def _handle_error(self, msg):
+        """
+        Reimplemented to reset the prompt if the error comes after the reply
+        """
+        super()._handle_error(msg)
```

Comment:
I think this is wrong. It has to be

    super(NamepaceBrowserWidget, self)._handle_error(msg)

Also, please move this addition to `ShellWidget` because it's not related to the Variable Explorer.